### DISCUSSION TITLE
EventEmitter.ANY_EVENT

### DISCUSSION THREAD

**Discussion by mike-shields-dev** [djipco/webmidi#234] [Q&A] [answered]
Is there a way to listen to all midi events on a particular port, channel or globally?  

I have found this page in the docs: [EventEmitter.ANY_EVENT](https://webmidijs.org/api/classes/WebMidi/#ANY_EVENT)

This sounds like what I want, but how do I use it?

**Comment by djipco** [comment, marked as answer]
There is currently no way to listen to all MIDI messages globally. You could, however, add a listener for `midimessage` to all `Input` or `InputChannel` objects.

This is how to do it at the `Input` level:

```js
WebMidi.inputs.forEach(input => {
  input.addListener("midimessage", e => console.log(e));
});
```

The difference between listening for [`midimessage` ](https://webmidijs.org/api/classes/Input#event-midimessage)and listening for [`EventEmitter.ANY_EVENT`](https://webmidijs.org/api/classes/EventEmitter#ANY_EVENT) is that the former is only triggered when a MIDI message is received while the latter is triggered by MIDI messages and other events such as `opened`, `closed` and `disconnected`.

**Reply by mike-shields-dev** [reply]
Thanks for clearing things up :+1:

**Reply by elmcapp** [reply]
```
WebMidi.inputs.forEach(input => {
  input.addListener("midimessage", e => console.log(e));
});
```
is not working at all for me. I was hope to add a listener for all message. I have to add the listener to everything. (note-on, note-off, cc, etc) I would like to listen to everything and the current code is not working for me. I am using the latest version of NodeJS
